src/test/java/org/mabrarov/exceptionsafety/FactoryMethodTest.java

```diff
@@ -41,9 +43,23 @@
 
   private OutputStream createConfiguredResource()
       throws IOException, TestResourceConfigurationException {
-    final OutputStream resource = createResource();
-    configureResource(resource);
-    return resource;
+    OutputStream resource = createResource();
+    Throwable throwable = null;
+    try {
+      configureResource(resource);
+      return resource;
+    } catch (TestResourceConfigurationException | RuntimeException|Error e) {
```

Comment:
Can't this line be simplified till (easy to copy-paste):

```suggestion
    } catch (Throwable e) {
```

?